sympy/polys/tests/test_hypothesis.py

```diff
@@ -41,3 +53,48 @@ def test_poly_hypothesis(coefficients1, coefficients2):
     g_q = Poly(coefficients2, x, domain="QQ")
     remainder_q = f_q.rem(g_q)
     assert g_q.degree() >= remainder_q.degree() or remainder_q.degree() == 0
+
+
+@given(
+    coefficients1=coefficients(),
+    coefficients2=coefficients(),
+)
+def test_multiplication(coefficients1, coefficients2):
+    f = Poly(coefficients1, x, domain="ZZ")
+    g = Poly(coefficients2, x, domain="ZZ")
+    h = f * g
+    assert h.degree() == f.degree() + g.degree()
+    assert h.LC() == f.LC() * g.LC()
+
+
+@given(
+    coefficients1=coefficients(),
+    coefficients2=coefficients(),
+)
+def test_addition(coefficients1, coefficients2):
+    f = Poly(coefficients1, x, domain="ZZ")
+    g = Poly(coefficients2, x, domain="ZZ")
+    h = f + g
+    if h.degree() != -float("inf"):
+        assert h.degree() == max(f.degree(), g.degree())
+
+
+@given(
+    coefficients1=coefficients(),
+    coefficients2=coefficients(empty=False),
```

Comment:
In general it would be better to just test general polynomials for all the functions. The test may have to have different branches based on whether something is 0 or not, but that is fine. Even if something gives an error, it's better to still generate it and test that the error is raised than to skip it. For example, here we can test that the lcm is always 0 if one of the arguments is 0. 

Also, if you write this test as `f*g == lcm(f, g)*gcd(f, g)` there would be no need to worry about zero division.

Reply:
The` empty=False`, is also to avoid  having both f,g be empty which causes a zero division error so `f*g == lcm(f, g)*gcd(f, g)` wouldn't bypass that. But yes, I can write  the test to expect an error when both polynomials are empty.

Reply:
@asmeurer Whats a nice way to test exceptions? I've tried figuring out ways to utilize `pytest.raises(ZeroDivisionError)` within `test_lcm` but with  no success.

Reply:
That's the correct way (except import `raises` from sympy.testing). Does that not work? If you search other test files you can find examples of how raises works.

Reply:
Yeah. When I try to call after a `with raises(ZeroDvisionError):` I  get either an `AttributeError: 'int' object has no attribute 'is_commutative'` or a seemingly nonsensical positional argument mismatch. I can't figure out  a workaround.

Reply:
You shouldn't try to call the test function recursively. 

lcm with 0 shouldn't produce an error. lcm(f, 0) should always produce 0. 

Algebraically, gcd and lcm form a bounded distributive lattice, with gcd as the "meet" and lcm as the "join". In this lattice, 1 is the bottom and 0 is the top. The axioms of a bounded distributive lattice are all things that could be tested in this function (for instance, `lcm(f, 0) == 0` corresponds to the lattice axiom $x \vee 1 = 1$).

Reply:
> I primarily get an AttributeError: 'int' object has no attribute 'is_commutative'

That's possibly a bug. What is the function call that produces that error?

Reply:
Here are the lattice axioms we should test

**Associativity**:
$(a ∨ b) ∨ c = a ∨ (b ∨ c)$
$(a ∧ b) ∧ c = a ∧ (b ∧ c)$

**Commutativity**:
$a ∨ b = b ∨ a$
$a ∧ b = b ∧ a$

**Idempotence**:
$a ∨ a = a$
$a ∧ a = a$

**Absorption**:
$a ∨ (a ∧ b) = a$
$a ∧ (a ∨ b) = a$

**Identity (Boundedness)**:
$a ∨ 0 = a$
$a ∧ 1 = a$
$a ∨ 1 = 1$
$a ∧ 0 = 0$

**Distributivity**:
$a ∧ (b ∨ c) = (a ∧ b) ∨ (a ∧ c)$
$a ∨ (b ∧ c) = (a ∨ b) ∧ (a ∨ c)$

Note that here $∧$ is gcd and $∨$ is lcm. And also note that $0$ and $1$ are reversed. $0$ corresponds to 1 and $1$ corresponds to 0. 

Here's a site that lists the lattice axioms https://math24.net/lattices.html. Note that gcd and lcm form a bounded distributive lattice, but some of the other types of lattice listed there like complemented lattice do not apply (modular does apply because every distributive lattice is modular, but testing that would be redundant).